pkg/sshd/server.go

```diff
@@ -225,7 +232,8 @@ func (s *server) answer(channel ssh.Channel, requests <-chan *ssh.Request, sshCo
 				}
 
 				repoName := parts[1]
-				if err := buildingRepos.Lock(repoName, time.Duration(0)); err != nil {
+				s.cleanerRef.Lock()
```

Comment:
This line will block the cleaner as well as *all other* `git push`es from proceeding. This preserves the goal of preventing multiple git pushes, but reverses the timeout functionality we introduced in #170